test/base/src/androidTest/kotlin/io/realm/test/shared/RealmConfigurationTests.kt

```diff
@@ -155,12 +197,22 @@ class RealmConfigurationTests {
     @Test
     fun name() {
         val realmName = "my.realm"
-
         val config = RealmConfiguration.Builder(schema = setOf(Sample::class)).name(realmName).build()
         assertEquals(realmName, config.name)
         assertTrue(config.path.endsWith(realmName))
     }
 
+    // TODO This should probably throw an IllegalArgumentException, for now just document behaviour.
```

Comment:
Maybe we should just not allow separators in the name at all?

Reply:
Yes, that would probably make sense...It should be easy to check for in the platform specific code, but it does mean the check is deferred to when the user is calling build() rather at the callsite...but that is probably acceptable until we get better file system support.